packages/mdc-ripple/index.js

```diff
@@ -63,10 +64,32 @@ class MDCRipple extends MDCComponent {
       isSurfaceDisabled: () => instance.disabled,
       addClass: (className) => instance.root_.classList.add(className),
       removeClass: (className) => instance.root_.classList.remove(className),
-      registerInteractionHandler: (evtType, handler) =>
-        instance.root_.addEventListener(evtType, handler, util.applyPassive()),
-      deregisterInteractionHandler: (evtType, handler) =>
-        instance.root_.removeEventListener(evtType, handler, util.applyPassive()),
+      registerInteractionHandler: (evtType, handler) => {
```

Comment:
At a glance it looks like WHATWG supports mouseup and keyup on the window, but not pointerup or touchend. I chose to use window and not documentElement or body because I believe that's the only way to get the event to fire if the mouse leaves the window boundaries. But being that pointerup and touchend are not supported on the window it might just make sense to ignore such a scenario rather than to split the "up events" (although I'm guessing here on how the project looks at such an issue).

Reply:
@kfranqueiro What do you think about provisionally adding the events to window for the browsers that support it, while also adding it to the documentElement to comply with spec? This way we can handle the cursor leaving the window boundaries on browsers that support it.

Reply:
Sorry for the slow response here. I'm reaching out to designers before answering this, because I had a thought about a possible alternative solution which might require less effort than managing document/window handlers, but I want to see which way seems correct to them first.

Reply:
Okay, I've received design feedback, and I think we are going down the correct path (in fact we may need to do more work in addition to what this PR addresses, but we can get to the rest later).

My thoughts (feel free to comment if it seems like I've left something out or messed up):

* I don't think keyup on the window should matter, since we're concerned with a bug related to events with pointing devices
* We will want 4 new adapter APIs: `registerWindowInteractionHandler`, `registerDocumentElementInteractionHandler`, and their `deregister` equivalents
* When a ripple is activated, we will register `mouseup` on window, and all `up` events on `documentElement`
  * May want to test whether we need `mouseup` on both, or whether `mouseup` on `window` is sufficient for all [supported browsers](https://github.com/material-components/material-components-web#browser-support)
* Within the `up` event callback, we will deregister all of the `up` event handlers

Again, there may be useful reference material for this in [mdc-slider's foundation](https://github.com/material-components/material-components-web/blob/bcc5ec5cc6f5ea9f1ca58baf03f2741d06768658/packages/mdc-slider/foundation.js#L248), though that simply listens for everything on `body`.